lib/settings.js

```diff
@@ -354,39 +358,45 @@ class Settings extends EventEmitter {
     return this;
   }
 
-  /**
-   * Returns the parsed contents of the settings file.
-   *
-   * @returns {Object}
-   * @private
-   */
-  _readSettings() {
-    this._ensureSettings();
-    try {
-      return jsonfile.readFileSync(this._getSettingsFilePath());
-    } catch (err) {
-      // Could not read the file. The user may not have permission to
-      // access the file or directory. Throw error.
-      throw err;
+  _tryReadAndParseJsonFile(filePath) {
+    const response = {
+      isOk: false,
+      content: '',
+      jsonObject: {}
+    };
+    if (fs.existsSync(filePath)) {
+      try {
+        response.content = fs.readFileSync(filePath, 'utf8');
+        response.jsonObject = JSON.parse(response.content);
+        response.isOk = true;
+      } catch (e) {
+        winston.error(e);
+        response.isOk = false; // not json format => need to write default setting
+        this.emit(Settings.Events.ERROR, {
+          error: e,
+          content: response.content
```

Comment:
why not just return all response?

Reply:
hm...not sure we need to return all or not because when the error occur below data is always fixed value
 - isValid = false
 - data = {} => because we can  not parse JSON

Reply:
I think it's fine, the user can decide how they want to handle raw/parsed data